app/code/Magento/Sales/Helper/Invoice/Logo.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Magento\Sales\Helper\Invoice;
+
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\UrlInterface;
+use Magento\Store\Model\ScopeInterface;
+
+class Logo extends AbstractHelper
+{
+    const XML_PATH_SALES_IDENTITY_LOGO_HTML = 'sales/identity/logo_html';
+
+    protected $logoBaseDir = 'sales/store/logo_html/';
+
+    /**
+     * @return string|null
+     */
+    public function getLogoFile()
+    {
+        $result = null;
+
+        $invoiceLogoPath = $this->getIdentityLogoHtml();
+        if ($invoiceLogoPath) {
+            $result = $this->_urlBuilder->getBaseUrl(
+                ['_type' => UrlInterface::URL_TYPE_MEDIA]
+            ) . $this->getLogoBaseDir() . $invoiceLogoPath;
+        }
+
+        return $result;
```

Comment:
I would recommend here to handle the negative case first:
```
        $invoiceLogoPath = $this->getIdentityLogoHtml();
        if (!$invoiceLogoPath) {
            return null;
        }
        return sprintf(
            "%s%s%s",
            $this->_urlBuilder->getBaseUrl(['_type' => UrlInterface::URL_TYPE_MEDIA]),
            $this->getLogoBaseDir(),
            $invoiceLogoPath
        );
```
Such a code seems cleaner for me, and we got rid of `$result` variable.

But it's just a recommendation. Feel free to use your approach.

Reply:
I agree.
My approach makes sense if I would do more checks, that can modify $result.